### PR TITLE
Refactor models to use new scm-base functions

### PR DESCRIPTION
**Pull request by tkyi**
- Use `scmUri` over `scmUrl`
- Call `parseUrl` to get `scmUri`
- Call `decorateCommit`
- Make changes to conform to new eslint rules
- Bump data-schema version to 14.0.0
- Bump models version to 16.0.0

Based on changes made in https://github.com/screwdriver-cd/scm-base/pull/11
Related to https://github.com/screwdriver-cd/screwdriver/issues/160

### CODE REVIEW

**Comment by FenrirUnbound** [comment]
Thing to note is that `scm-github` hasn't adopted the new `scm-base@2.x` interface

**Comment by tkyi** [comment]
@FenrirUnbound Should be okay if I bumped major versions for `models`, right?

File: lib/buildFactory.js [hunk 1]
```diff
@@ -117,20 +122,21 @@ class BuildFactory extends BaseFactory {
...
-                    .then(sha => {
+                return getCommitInfo({ job, scm: this.scm, modelConfig })
+                    .then((data) => {
                         // TODO: support matrix jobs
                         const index = number.toString().split('.')[1] || 0;
                         const permutation = job.permutations[index];
 
-                        modelConfig.sha = sha;
+                        modelConfig.commit = data;
+                        console.log('commit info: ', modelConfig.commit);
```
Comment (petey) [inline lib/buildFactory.js]: console.log

**Review by petey — CHANGES_REQUESTED** [review]

File: lib/buildFactory.js [hunk 2]
```diff
@@ -44,9 +39,19 @@ function getCommitSha(config) {
 
         return user.unsealToken()
             .then(token => config.scm.getCommitSha({
-                scmUrl: pipeline.scmUrl,
+                scmUri: pipeline.scmUri,
                 token
-            }));
+            })
+            .then((sha) => {
+                config.modelConfig.sha = sha;
```
Comment (d2lam) [inline lib/buildFactory.js]: What is this for? It's not being used.

File: lib/pipeline.js [hunk 3]
```diff
@@ -276,44 +278,50 @@ class PipelineModel extends BaseModel {
...
-        // get the admin of the pipeline
-        return this.admin
-            // and their token
-            .then(user => user.unsealToken())
+        return Promise.all([
+            this.admin,                          // get the admin of the pipeline
+            this.scm.parseUrl(this.checkoutUrl)  // get the scmUri
+        ]).then(([user, scmUri]) => {
+            this.scmUri = scmUri;
+            this.checkoutUrl = null; // TODO: delete checkoutUrl?
```
Comment (d2lam) [inline lib/pipeline.js]: Why?

**Review by d2lam — CHANGES_REQUESTED** [review]

File: lib/buildFactory.js (hunk 2 above)
Comment (tkyi) [reply lib/buildFactory.js]: Do we not want to return the `sha` in the `build` blob?

File: lib/pipeline.js (hunk 3 above)
Comment (tkyi) [reply lib/pipeline.js]: I don't think we want to save `checkoutUrl` permanently in the `pipeline` blob

File: lib/buildFactory.js (hunk 2 above)
Comment (d2lam) [reply lib/buildFactory.js]: What do you mean? You're just setting the config.modelConfig.sha here. You're not returning or doing anything with it in this function.

Comment (FenrirUnbound) [reply lib/buildFactory.js]: It looks like we're modifying a parameter, which is a code smell. The function already passes back the SHA, as well as additional data regarding the SHA. 

Why do we want the function to also mutate a given parameter?

Comment (tkyi) [reply lib/buildFactory.js]: The `decorateCommit` function does not include a SHA in its return.
Changed the `getCommitInfo` function to return an object with the decorated commit and the SHA.

**Review by d2lam — APPROVED** [review]
LGTM 👍

**Review by petey — APPROVED** [review]

**Pull request merged**